src/EventGrid/EventGrid/SystemTopic/NewAzureEventGridSystemTopic.cs

```diff
@@ -0,0 +1,209 @@
+        [ValidateNotNullOrEmpty]
+        [Alias("ResourceType")]
+        public string ResourceType { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 6,
+            HelpMessage = EventGridConstants.TopicLocationHelp,
+            ParameterSetName = SystemTopicNameParameterSet)]
+        [LocationCompleter("Microsoft.EventGrid/systemTopics")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        /// <summary>
+        /// string which represents the source.
+        /// </summary>
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.TagsHelp,
+            ParameterSetName = SystemTopicNameParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string Source { get; set; }
+
+        /// <summary>
+        /// string which represents the topic type.
+        /// </summary>
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.TagsHelp,
+            ParameterSetName = SystemTopicNameParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string TopicType { get; set; }
+
+        /// <summary>
+        /// string which represents the MetricResourceId.
+        /// </summary>
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.TagsHelp,
+            ParameterSetName = SystemTopicNameParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string MetricResourceId { get; set; }
+
+        /// <summary>
+        /// string which represents the IdentityType.
+        /// </summary>
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.TagsHelp,
+            ParameterSetName = SystemTopicNameParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string IdentityType { get; set; }
+
+        /// <summary>
+        /// string which represents the PrincipalId.
+        /// </summary>
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.TagsHelp,
+            ParameterSetName = SystemTopicNameParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string PrincipalId { get; set; }
+
+        /// <summary>
+        /// string which represents the TenantId.
+        /// </summary>
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.TagsHelp,
+            ParameterSetName = SystemTopicNameParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string TenantId { get; set; }
```

Comment:
We don't support targeting to a specified TenantId in Azure Cmdlet. We usually switch tenant by Set-AzContext -TenantId "xxx"

Reply:
fixed